components/applications-service/integration_test/ingestion_test.go

```diff
@@ -5,4 +5,277 @@
+			withEnvironment("development"),
+			withFqdn("app.example.com"),
+			withSite("us"),
+		),
+		NewHabitatEvent(
+			withSupervisorId("4f1un3"),
+			withPackageIdent("test/web/0.1.0/20200101121212"),
+			withServiceGroup("web.default"),
+			withStrategyAtOnce("stable"),
+			withApplication("test-app"),
+			withEnvironment("development"),
+			withFqdn("web.example.com"),
+			withSite("us"),
+		),
+		NewHabitatEvent(
+			withSupervisorId("m4y4"),
+			withPackageIdent("test/db/0.1.0/20200101121213"),
+			withServiceGroup("db.default"),
+			withStrategyAtOnce("stable"),
+			withApplication("test-app"),
+			withEnvironment("development"),
+			withFqdn("db.example.com"),
+			withSite("us"),
+		),
+	}
+
+	// Ingest messages through the Ingester client
+	suite.IngestMessagesViaIngester(events...)
+
+	// Verify there are only three services
+	svcList := suite.GetServices()
+	assert.Equal(t, len(events), len(svcList), "wrong number of services")
+
+	// Verify there are only three service_groups
+	sgList := suite.GetServiceGroups()
+	assert.Equal(t, 3, len(sgList), "wrong number of service groups")
+}
+
+func TestIngestMultiServicesSameServiceGroup(t *testing.T) {
+	defer suite.DeleteDataFromStorage()
+
+	events := []*habitat.HealthCheckEvent{
+		NewHabitatEvent(
+			withSupervisorId("s4l1m"),
+			withPackageIdent("test/db/0.1.0/20200101121211"),
+			withServiceGroup("db.default"),
+			withStrategyAtOnce("stable"),
+			withApplication("test-app"),
+			withEnvironment("development"),
+			withFqdn("db.example.com"),
+			withHealth(HealthCheckIntToString(0)), // -> OK
+			withSite("us"),
+		),
+		NewHabitatEvent(
+			withSupervisorId("4f1un3"),
+			withPackageIdent("test/db/0.1.0/20200101121212"),
+			withServiceGroup("db.default"),
+			withStrategyAtOnce("stable"),
+			withApplication("test-app"),
+			withEnvironment("development"),
+			withFqdn("db.example.com"),
+			withHealth(HealthCheckIntToString(0)), // -> OK
+			withSite("us"),
+		),
+		NewHabitatEvent(
+			withSupervisorId("m4y4"),
+			withPackageIdent("test/db/0.1.0/20200101121213"),
+			withServiceGroup("db.default"),
+			withStrategyAtOnce("stable"),
+			withApplication("test-app"),
+			withEnvironment("development"),
+			withFqdn("db.example.com"),
+			withHealth(HealthCheckIntToString(0)), // -> OK
+			withSite("us"),
+		),
+	}
+
+	// Ingest messages through the Ingester client
+	suite.IngestMessagesViaIngester(events...)
+
+	// Verify there are only three services
+	svcList := suite.GetServices()
+	assert.Equal(t, len(events), len(svcList), "wrong number of services")
+
+	// We have a race condition where if we have different services sending messages
+	// to Automate at the same time that belongs to the same service-group we will
+	// insert different service-groups and deployments entries instead of a single one.
+	//
+	// The main reason of this problem is because we process 50 things at a time (worker pool)
+	//
+	// TODO @afiune fix and uncomment this test!
```

Comment:
This is mentioned in the PR description... did you fix it or just find it?

Reply:
I found it, I didn't know if the PR would get too big for review so I first added tests to discover the failures. Though I have already fixed a number of them and pushed my fixes, let me know what you think? 

I didn't fix the race condition, I am going to write a bug card to fix it in a different PR. 💯

Reply:
Jira bug card: https://chefio.atlassian.net/browse/A2-890